<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>2.2.1. G-OnRamp Features &#8212; G-OnRamp 0.1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="2.3.1. G-OnRamp Tools" href="Tools.html" />
    <link rel="prev" title="2.1.1. What is G-OnRamp" href="What_is_G-OnRamp.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body role="document">
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="g-onramp-features">
<h1>2.2.1. G-OnRamp Features<a class="headerlink" href="#g-onramp-features" title="Permalink to this headline">¶</a></h1>
<div class="section" id="web-based-platform">
<h2>2.2.1.1. Web-based platform<a class="headerlink" href="#web-based-platform" title="Permalink to this headline">¶</a></h2>
<p>G-OnRamp provides an integrated, web-based, graphical interface for performing large scale bioinformatics analyses. G-OnRamp is based on Galaxy platform. Galaxy (<a class="reference external" href="http://galaxyproject.org/">http://galaxyproject.org/</a>, <a class="reference external" href="https://usegalaxy.org">https://usegalaxy.org</a>) is an open-source, web-based scientific gateway for accessible, reproducible, and transparent analyses of large biomedical datasets that is used throughout the world. G-OnRamp extends Galaxy with analysis workflows that create a graphical genome browser for annotation, including evidence from sequence homology, gene predictions, and RNA-seq. The end users do not need to install any software and no programming skills are required. The users can access the G-OnRamp workflow and run the analysis through a graphical interface to create genome browsers for new genome assemblies.</p>
</div>
<div class="section" id="designed-for-any-genome">
<h2>2.2.1.2. Designed for any genome<a class="headerlink" href="#designed-for-any-genome" title="Permalink to this headline">¶</a></h2>
<p>G-OnRamp is designed to work with any eukaryotic genomes. The 10 participants from 9 institutions to the beta testers workshop (July 26-28, 2016) worked on five genome assemblies: Kryptolebias marmoratus, Chlamydomonas reinhardtii, Amazona vittata, Sebastes rubrivinctus, Xenopus laevis. The <a class="reference external" href="https://de.cyverse.org/anon-files/iplant/home/shared/G-OnRamp_hubs/index.html">assembly hubs</a> of these species can be viewed at UCSC Genome Browser.</p>
<img alt="../_images/species.jpg" src="../_images/species.jpg" />
</div>
<div class="section" id="reproducible-workflow">
<h2>2.2.1.3. Reproducible workflow<a class="headerlink" href="#reproducible-workflow" title="Permalink to this headline">¶</a></h2>
<p>G-OnRamp record all the steps in the analysis workflow (e.g., tool parameters, data conversions) so that you can apply the same workflow to other eukaryotic genomes. All the datasets, including uploaded datasets and the output datasets from each steps that generate a dataset, are stored by Galaxy in Histories. The tools and their settings used in each step are also stored in the Histories. For more information, see <a href="#id1"><span class="problematic" id="id2">`Galaxy Histories &lt;&gt;`_</span></a>. An analysis workflow can be generated from the analysis already completed in a history. It contains a series of tools and settings used in each step, which makes it simple to applying the same methods to the other datasets. For more information, see <a href="#id3"><span class="problematic" id="id4">`Extracting Galaxy Workflow &lt;&gt;`_</span></a> from a History.</p>
</div>
<div class="section" id="share-your-results">
<h2>2.2.1.4. Share your results<a class="headerlink" href="#share-your-results" title="Permalink to this headline">¶</a></h2>
<p>The G-OnRamp analysis history (datasets, tools, and parameter settings) can be shared publicly or only with your colleagues. The users can share their Galaxy items, including histories and workflows, with other people in three ways: share with individual users, share with one or more users, and share with everyone. The video below shows how to share and publish your Galaxy items:</p>
<iframe src="https://player.vimeo.com/video/75934770?color=0fbe7c&portrait=0" width="640" height="360" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe></div>
<div class="section" id="customizable">
<h2>2.2.1.5. Customizable<a class="headerlink" href="#customizable" title="Permalink to this headline">¶</a></h2>
<p>You can incorporate additional Galaxy tools and modify the G-OnRamp workflow (e.g., tool parameters) according to your needs. The easiest way to get new tools is to install them from the Tool Shed. The Add Tool From Tool Shed Tutorial describes step by step instructions about how to install a tool from the Tool Shed. In some cases, the workflow needs to be adjusted (e.g. change the parameters settings of some tools) for running the analysis on the new datasets. Basic Workflow Editing provides a tutorial of how to perform basic workflow editing. The Customize the Genome Browsers produced by G-OnRamp is a walkthrough of how to edit a G-OnRamp workflow.</p>
</div>
<div class="section" id="multiple-deployment-options">
<h2>2.2.1.6. Multiple deployment options<a class="headerlink" href="#multiple-deployment-options" title="Permalink to this headline">¶</a></h2>
<p>G-OnRamp analysis can be done on the cloud (e.g., Amazon EC2) or on the a local virtual machine (e.g., VirtualBox) instance.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">2.2.1. G-OnRamp Features</a><ul>
<li><a class="reference internal" href="#web-based-platform">2.2.1.1. Web-based platform</a></li>
<li><a class="reference internal" href="#designed-for-any-genome">2.2.1.2. Designed for any genome</a></li>
<li><a class="reference internal" href="#reproducible-workflow">2.2.1.3. Reproducible workflow</a></li>
<li><a class="reference internal" href="#share-your-results">2.2.1.4. Share your results</a></li>
<li><a class="reference internal" href="#customizable">2.2.1.5. Customizable</a></li>
<li><a class="reference internal" href="#multiple-deployment-options">2.2.1.6. Multiple deployment options</a></li>
</ul>
</li>
</ul>
<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
  <li><a href="../introduction.html">2. Introduction of G-OnRamp</a><ul>
      <li>Previous: <a href="What_is_G-OnRamp.html" title="previous chapter">2.1.1. What is G-OnRamp</a></li>
      <li>Next: <a href="Tools.html" title="next chapter">2.3.1. G-OnRamp Tools</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/introduction/Features.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2016, Yating Liu.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.5.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.9</a>
      
      |
      <a href="../_sources/introduction/Features.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>