<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>2.4.1. Workflows &#8212; G-OnRamp 0.1.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="3. Tutorials of G-OnRamp" href="../tutorials.html" />
    <link rel="prev" title="2.3.1. G-OnRamp Tools" href="Tools.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body role="document">
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="workflows">
<h1>2.4.1. Workflows<a class="headerlink" href="#workflows" title="Permalink to this headline">¶</a></h1>
<div class="section" id="big-picture">
<h2>2.4.1.1. Big picture<a class="headerlink" href="#big-picture" title="Permalink to this headline">¶</a></h2>
<p>We have developed a comprehensive Galaxy workflow that produces multiple complementary datasets to facilitate the annotation of any eukaryotic genome. The entire workflow is shown below.</p>
<img alt="../_images/Big_picture.png" src="../_images/Big_picture.png" />
</div>
<div class="section" id="sub-workflows">
<h2>2.4.1.2. Sub-workflows<a class="headerlink" href="#sub-workflows" title="Permalink to this headline">¶</a></h2>
<p>The G-OnRamp workflow is divided into four sub-workflows: homology, repeat regions, RNA-Seq, and gene predictions. These sub-workflows will produce the input datasets for the Hub Archive Creator, which will create the UCSC Genome Browser Assembly Hub.</p>
<div class="section" id="homology">
<h3>2.4.1.2.1. Homology<a class="headerlink" href="#homology" title="Permalink to this headline">¶</a></h3>
<p>The genome assembly (in FASTA format) is the input dataset for the NCBI BLAST+ tool makeblastdb, which creates a nucleotide database for BLAST searches. The NCBI BLAST+ tblastn tool searches this nucleotide database against a collection of protein query sequences from an informant species. The blastXmlToPsl and pslToBed tools are used to convert the tblastn search results to the BED format required by the Hub Archive Creator.</p>
</div>
<div class="section" id="repeat-regions">
<h3>2.4.1.2.2. Repeat regions<a class="headerlink" href="#repeat-regions" title="Permalink to this headline">¶</a></h3>
<p>TrfBig partitions the genome assembly into smaller chunks and then runs Tandem Repeats Finder (TRF) on each chunk to identify tandem repeats within each genomic region. Note that the output of TRF is in BED4+12 format.</p>
</div>
<div class="section" id="rna-seq">
<h3>2.4.1.2.3. RNA-seq<a class="headerlink" href="#rna-seq" title="Permalink to this headline">¶</a></h3>
<p>RNA-Seq reads are mapped against the genome assembly by HISAT2, and StringTie assembles the mapped RNA-Seq reads into potential transcripts. The “junctions extract” subprogram in Regtools reports the locations of putative introns based on the spliced RNA-Seq reads in the BAM file. The RNA-Seq read coverage track was created by the “Convert BAM to BigWig” tool.</p>
</div>
<div class="section" id="gene-predictions">
<h3>2.4.1.2.4. Gene predictions<a class="headerlink" href="#gene-predictions" title="Permalink to this headline">¶</a></h3>
<p>Gene models from three gene predictors (Augustus, GlimmerHMM, and SNAP) were produced using species-specific parameters if they were available. The gene prediction results are converted into the bigGenePred format by the Hub Archive Creator.</p>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">2.4.1. Workflows</a><ul>
<li><a class="reference internal" href="#big-picture">2.4.1.1. Big picture</a></li>
<li><a class="reference internal" href="#sub-workflows">2.4.1.2. Sub-workflows</a><ul>
<li><a class="reference internal" href="#homology">2.4.1.2.1. Homology</a></li>
<li><a class="reference internal" href="#repeat-regions">2.4.1.2.2. Repeat regions</a></li>
<li><a class="reference internal" href="#rna-seq">2.4.1.2.3. RNA-seq</a></li>
<li><a class="reference internal" href="#gene-predictions">2.4.1.2.4. Gene predictions</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
  <li><a href="../introduction.html">2. Introduction of G-OnRamp</a><ul>
      <li>Previous: <a href="Tools.html" title="previous chapter">2.3.1. G-OnRamp Tools</a></li>
      <li>Next: <a href="../tutorials.html" title="next chapter">3. Tutorials of G-OnRamp</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/introduction/Workflows.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2016, Yating Liu.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.5.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.9</a>
      
      |
      <a href="../_sources/introduction/Workflows.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>